python/langsmith/client.py

```diff
@@ -931,6 +933,43 @@ def request_with_retries(
                     filler = "*" * (max(0, len(api_key) - 7))
                     masked_api_key = f"{prefix}{filler}{suffix}"
 
+                    debug_crash_dump_file = ls_utils.get_env_var(
+                        "LANGSMITH_DEBUG_CRASH_DUMP"
+                    )
+                    if debug_crash_dump_file is not None and e.request is not None:
+                        try:
+                            headers = e.request.headers.copy()
```

Comment:
@hinthornw Let me know if you agree with this approach.

Made a shallow copy of the headers so we could get rid of the x-api-key, but log the rest of the headers.

Manually added all of the fields which I think are relevant here. This is to prevent us accidentally logging other sensitive fields (if any).